UnityProject/Assets/Scripts/Machines/Microwave.cs

```diff
@@ -104,9 +116,28 @@ private void FinishCooking()
 		if (isServer)
 		{
 			GameObject mealPrefab = CraftingManager.Meals.FindOutputMeal(meal);
-			Spawn.ServerPrefab(mealPrefab, GetComponent<RegisterTile>().WorldPosition, transform.parent);
+			SpawnResult result = Spawn.ServerPrefab(mealPrefab, GetComponent<RegisterTile>().WorldPosition, transform.parent);
+			
+			//If the resulting meal has a stackable component, set the amount to mealCount to ensure that food in = food out.
+			Stackable stck = result.GameObject.GetComponent<Stackable>();
+
+			if (stck != null && mealCount != 0)
+			{
+				//Get difference between new item's initial amount and the amount held by mealCount (amount of ingredient).
+				//If not 0, add the difference to make new item amount equal to old item amount.
+				int stckChanger = mealCount-stck.Amount;
+
+				if (stckChanger != 0)
+				{
+					stck.ServerConsume(-stckChanger);
```

Comment:
I don't think ServerConsume is designed to work with negative values. Perhaps you could change `stckChanger != 0` to `stckChanger > 0` to make sure it's positive when inverted?

Reply:
Is there another way to increase the stack amount of a newly stacked item? ServerConsume was implemented this way in order to ensure that a stackable item out has the same amount as the stackable item in (if applicable). The initial spawn amount for a cooked cutlet, for example, is 3. Both raw and cooked cutlets have a max size of 6. If I put in 4 raw cutlets, I would need to add 1 more to the cooked item amount, whereas if I put in 2, I'd have to subtract 1 from the product stack. ServerConsume can handle the second case, but I'd need another way to handle the first.

Right now, last time I checked, the only way to directly adjust the amount in a stackable item is through SyncAmount, a private function. Another option would be to spawn more stacks as needed of an object and use ServerCombine to add them all up into a single entity, and that seems like a bit more work than needed.

Reply:
Decided to just make a ServerIncrease function for Stackable and had microwave implement it.